pkg/kubelet/kubelet_pods.go

```diff
@@ -1208,3 +1243,42 @@ func (kl *Kubelet) PortForward(podFullName string, podUID types.UID, port uint16
 	}
 	return kl.runner.PortForward(&pod, port, stream)
 }
+
+// cleanupOrphanedPodCgroups removes the Cgroups of pods that should not be
+// running and whose volumes have been cleanedup.
+func (kl *Kubelet) cleanupOrphanedPodCgroups(
+	pods []*api.Pod, runningPods []*kubecontainer.Pod) error {
+	// Add all running and existing terminated pods to a set allPods
+	allPods := sets.NewString()
+	for _, pod := range pods {
+		allPods.Insert(string(pod.UID))
+	}
+	for _, pod := range runningPods {
+		allPods.Insert(string(pod.ID))
+	}
+
+	pcm := kl.containerManager.NewPodContainerManager()
+	found, err := pcm.GetAllPodsFromCgroups()
```

Comment:
@vishh -- i think this logic is prone to race conditions and may be the reason why e2e tests were flaky.

1. allPods, runningPods = ...
1. kubelet creates a new pod A
1. list all pods based on state of cgroup tree
1. A is not in the initial allPods, runningPods list, so we think its orphaned and destroy the group+pid prematurely?

Am I mistaking the potential for a race condition here?